app/models/app_feature_completion_migration.rb

```diff
@@ -0,0 +1,57 @@
+class AppFeatureCompletionMigration
+  include ActiveModel::Validations
+
+  attr_reader :app, :feature_gate, :version_before_type_cast
+
+  validate :feature_gate_must_exist
+
+  delegate :version, to: :app_feature_completion
+
+  def initialize(params)
+    @app = params[:app] || raise("Must provide app")
+    @feature_gate = params[:feature_gate]
+    app_feature_completion.version = @version_before_type_cast = params[:version]
```

Comment:
I'm having trouble figuring out why you can't `delegate :version_before_type_cast` too. Mind illuminating?

Reply:
i spose I could, I just would have to have confidence in my understanding of the nuances of the implementation of ARBTC, when I really don't need a nuanced implementation here - I was looking for a revealing way to express that I care about whether I was handed a blank one of these or not regardless of what happens in my downstream.

Reply:
Fair. I think I'd trust it to fit the needs of this class, especially if the tests accurately describe the behavior in question ("i pass an empty version string") without stubbing the BTC value or anything. But I buy that this puts it front and center for someone trying to figure out how the destroy case gets triggered. I might in that case rename it to `@destroy` and read directly from that within `destroy?` (or something to avoid name conflicts on attr names -- `destroy_record`?) to make that a much tighter loop.

Reply:
I'm trying to reference some common language about what's going on by naming it the same thing that AR would name it. I can't just set `@destroy` to the BTC value, I'd need to call .present? on the BTC value to make it behave correctly, and right now that resides in the destroy method, making it more than just a delegation.